### DISCUSSION TITLE
Integrating DistilGPT-2 into UAgent Ecosystem

### DISCUSSION THREAD

**Discussion by gaurav19908** [fetchai/uAgents#129] [Integrations]
Hey Fetch Family! :tada:
I was looking into Distil, and I had an idea. I thought it would be cool to integrate DistilGPT-2 into uagent ecosystem. It is nothing is a distilled version of the GPT-2 model, has pretty much the same capabilities but with lighter computational power. This could potentially enhance the NLP capabilities for the uagents. More specifically, it could help with:
*Better Communication: uagents can chat with users as naturally, and in a more 'human-like' manner.
*Better content creation: Lot of use cases require drafting clean content - it can do that.
*Learning and adaptation: Over time, the agent could learn from the user’s interactions and refine its responses accordingly.
The key idea here is to get better speed (I know it is GPT-2, but it is still SUPER lightweight)and adaptibility (can give users more tailored experience when they have a conversation)
I am almost done with the code - and I think I will open a PR soon. lmk!

**Comment by rishankjhavar** [comment]
This is very promising! Please feel free to raise a PR - we'll try to review it as soon as possible.

Thank you so much for your efforts! 🚀

**Comment by abhifetch** [comment]
Hi @gaurav19908 , 
we have distilgpt included here in integrations. https://github.com/fetchai/uAgents/tree/b83a0a2a196478d0883f951b78bbe5dbdcea30cb/integrations/distilgpt2

if you have any comments on it, please feel free to comment here.
